src/control/lib/netdetect/netdetect.go

```diff
@@ -1224,3 +1231,61 @@ func ScanFabric(provider string, excludes ...string) ([]FabricScan, error) {
 	}
 	return ScanResults, nil
 }
+
+// GetDeviceClassStub is used for unit testing to replace GetDeviceClass where the network device
+// is not backed by physical hardware resources and does not have a corresponding entry in the file system.
+func GetDeviceClassStub(netdev string) (int32, error) {
+	return 0, nil
+}
+
+// GetDeviceClass determines the device type according to what's stored in the filesystem
+// Returns an integer value corresponding to its ARP protocol hardware identifier
+// found here: https://elixir.free-electrons.com/linux/v4.0/source/include/uapi/linux/if_arp.h#L29
+func GetDeviceClass(netdev string) (int32, error) {
+	devClass, err := ioutil.ReadFile(fmt.Sprintf("/sys/class/net/%s/type", netdev))
+	if err != nil {
+		return 0, err
+	}
+
+	res, err := strconv.Atoi(strings.TrimSpace(string(devClass)))
+	return int32(res), err
+}
+
+// Convert a network device class ID to a string identifier according to the table
+// found here: https://elixir.free-electrons.com/linux/v4.0/source/include/uapi/linux/if_arp.h#L29
+func DevClassName(class int32) string {
```

Comment:
Seeing this I think the device class should be its own type with possible values defined as an enum. This function could be turned into a String() method on that type.

Reply:
I made a version that a NetworkDeviceClass type, but that version felt klunky to me compared with passing a uint32 around.  I've left it as uint32 for now.  If we can agree to keep it that way, that is my first choice.  If not, I'll make a new type.